saiph/reduction/famd.py

```diff
@@ -401,9 +402,10 @@ def compute_categorical_cos2(
     -------
         dataframe of categorical cos2
     """
-    if model.U is not None and model.s is not None:
+    df_model = fit(df=df)
+    if df_model.U is not None and df_model.s is not None:
         model_coords = pd.DataFrame(
-            model.U[:, :min_nf] * model.s[:min_nf],
+            df_model.U[:, :min_nf] * df_model.s[:min_nf],
```

Comment:
if we want to compute cos2 values of a df based on a model fitted on another df, coords need to be computed df based not model based.